### PR TITLE
Improve timing for gtx86 and update plotting

### PR DESCRIPTION
**Pull request by twicki**
## Purpose

This PR improves how we measure timings for the gtx86 backend on daint and improves the plotting script:

## Code changes:

- Use 12 threads per rank, no hyperthreading and 1 rank per node on daint
- As a result we switch to the normal partition for this run
- In order to ensure compatibility with the fortran generated performance numbers we now also write out the hit-counts
- Fixes the key values to adapt the new naming 
- Plotting time per timestep for the main loop
- 
## Checklist
Before submitting this PR, please make sure:

- [X] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [X] Docstrings and type hints are added to new and updated routines, as appropriate

### CODE REVIEW

File: examples/standalone/benchmarks/plotting.py [hunk 1]
```diff
@@ -30,23 +30,34 @@
...
                         [
                             datetime.strptime(
                                 e["setup"]["timestamp"], "%d/%m/%Y %H:%M:%S"
                             )
                             for e in specific
                         ],
-                        [e["times"][key]["mean"] for e in specific],
-                        label=key + " " + backend,
+                        [
+                            e["times"][line]["mean"]
```
Comment (oelbert) [inline examples/standalone/benchmarks/plotting.py:51]: What does e mean? Element?

Comment (twicki) [reply examples/standalone/benchmarks/plotting.py:51]: yes. I tend to use 1-letter abbreviations when doing list comprehensions for readabiliy

Comment (oelbert) [reply examples/standalone/benchmarks/plotting.py:51]: To me this is really unclear and I would like to expand it to a word

**Review by oelbert — CHANGES_REQUESTED** [review]

**Review by oelbert — APPROVED** [review]
Good stuff

File: examples/standalone/benchmarks/plotting.py [hunk 2]
```diff
@@ -30,40 +30,80 @@
         key=lambda k: datetime.strptime(k["setup"]["timestamp"], "%d/%m/%Y %H:%M:%S")
     )
     for plottype in ["mainLoop", "initializationTotal"]:
-        keyval = (
-            ["main_loop"] if plottype == "mainLoop" else ["initialization", "total"]
+        data = (
```
Comment (ofuhrer) [inline examples/standalone/benchmarks/plotting.py:33]: Would it make sense to move this on-the-fly definition to a CONSTANT at the beginning of this script with a comment, moving in the direction of making this configurable.

File: examples/standalone/benchmarks/plotting.py [hunk 3]
```diff
@@ -30,40 +30,80 @@
...
+            ["mainloop", "DynCore", "Remapping", "TracerAdvection"]
+            if plottype == "mainLoop"
+            else ["initialization", "total"]
         )
         plt.figure()
         for backend in ["python/gtx86", "python/numpy", "fortran", "python/gtcuda"]:
```
Comment (ofuhrer) [inline examples/standalone/benchmarks/plotting.py:39]: Same here (see configuration comment above).

File: examples/standalone/benchmarks/plotting.py (hunk 2 above)
Comment (ofuhrer) [reply examples/standalone/benchmarks/plotting.py:33]: Is this `data` or `timers`?

File: examples/standalone/benchmarks/plotting.py [hunk 4]
```diff
@@ -30,40 +30,80 @@
...
             specific = [x for x in alldata if x["setup"]["version"] == backend]
             if specific:
-                for key in keyval:
+                for line in data:
```
Comment (ofuhrer) [inline examples/standalone/benchmarks/plotting.py:42]: `timer`?

File: examples/standalone/benchmarks/run_on_daint.sh [hunk 5]
```diff
@@ -79,10 +79,11 @@ echo "    Slurm output dir: $ROOT_DIR"
 cp buildenv/submit.daint.slurm .
 sed s/\<NAME\>/standalone/g submit.daint.slurm -i
 sed s/\<NTASKS\>/$ranks/g submit.daint.slurm -i
-sed s/\<NTASKSPERNODE\>/$ranks/g submit.daint.slurm -i
-sed s/\<CPUSPERTASK\>/1/g submit.daint.slurm -i
-sed s/#SBATCH\ --output=\<OUTFILE\>//g submit.daint.slurm -i
+sed s/\<NTASKSPERNODE\>/1/g submit.daint.slurm -i
+sed s/\<CPUSPERTASK\>/12/g submit.daint.slurm -i
```
Comment (ofuhrer) [inline examples/standalone/benchmarks/run_on_daint.sh:83]: You can get the 12 out of the `machineEnvironment.sh` afaik.

Comment (ofuhrer) [reply examples/standalone/benchmarks/run_on_daint.sh:83]: Should you not also set `#SBATCH --cpus-per-task=12` and add `export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK` to the CMD to enable OpenMP thread placement?

File: examples/standalone/runfile/dynamics.py [hunk 6]
```diff
@@ -49,7 +49,7 @@ def set_experiment_info(experiment_name, time_step, backend):
             pathlib.Path(__file__).parent.absolute(), search_parent_directories=True
         ).head.object.hexsha
     except git.InvalidGitRepositoryError:
-        sha = "6ac5225202279ae3ff2e5bebb224b817"
+        sha = "hash not found"
```
Comment (ofuhrer) [inline examples/standalone/runfile/dynamics.py:52]: Good catch!

**Review by ofuhrer — CHANGES_REQUESTED** [review]
A couple of minor change suggestions. Thanks for pushing this forward.

File: examples/standalone/benchmarks/run_on_daint.sh (hunk 5 above)
Comment (twicki) [reply examples/standalone/benchmarks/run_on_daint.sh:83]: The template has the line `export OMP_NUM_THREADS=<CPUSPERTASK>` already in there, we sed both at the same time

Comment (twicki) [reply examples/standalone/benchmarks/run_on_daint.sh:83]: the buildenv sets it to 8. Do we know why?

Comment (ofuhrer) [reply examples/standalone/benchmarks/run_on_daint.sh:83]: Oh gosh. Not a clue. Possibly abused for `make -j ${nthreads}` and thus limited to a reasonable number? Conclusion from my side: don't use `machineEnvironment.sh`.

Comment (twicki) [reply examples/standalone/benchmarks/run_on_daint.sh:83]: ok

File: examples/standalone/benchmarks/run_on_daint.sh [hunk 7]
```diff
@@ -79,10 +81,11 @@ echo "    Slurm output dir: $ROOT_DIR"
 cp buildenv/submit.daint.slurm .
 sed s/\<NAME\>/standalone/g submit.daint.slurm -i
 sed s/\<NTASKS\>/$ranks/g submit.daint.slurm -i
-sed s/\<NTASKSPERNODE\>/$ranks/g submit.daint.slurm -i
-sed s/\<CPUSPERTASK\>/1/g submit.daint.slurm -i
-sed s/#SBATCH\ --output=\<OUTFILE\>//g submit.daint.slurm -i
+sed s/\<NTASKSPERNODE\>/1/g submit.daint.slurm -i
+sed s/\<CPUSPERTASK\>/nthreads/g submit.daint.slurm -i
```
Comment (ofuhrer) [inline examples/standalone/benchmarks/run_on_daint.sh:85]: `$nthreads`?

Comment (ofuhrer) [reply examples/standalone/benchmarks/run_on_daint.sh:85]: Possibly move `-i` in front for these sed commands, as is customary?

**Review by ofuhrer — APPROVED** [review]
LGTM! One possible bug and a minor suggestion.

File: examples/standalone/benchmarks/run_on_daint.sh (hunk 7 above)
Comment (twicki) [reply examples/standalone/benchmarks/run_on_daint.sh:85]: is it? I am the worst basher - open to any suggestions here

**Pull request merged**